core/src/main/java/org/apache/iceberg/io/ContentCacheManager.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.io;
+
+/**
+ * Manages content caches for FileIO implementations, capable of creating or retrieving
+ * a {@link ContentCache} given a {@link FileIO}, or dropping a cache for a {@link FileIO}.
+ */
```

Comment:
Sorry not sure if I completely follow, why is `ContentCacheManager` required? Couldn't we achieve the desired goals with just having a pluggable `ContentCache`?

Reply:
The reason for additionally adding a `ContentCacheManager`, is to override the cache key.  The current default in `ManifestFiles` presumes a single long lived instance of a `FileIO`.  However, users of custom FileIOs may not make this presumption.  In order to accommodate that, you can override the `ContentCacheManager`.  The alternative might be to specify a custom key in the `FileIO` itself.  Additionally, by introducing a manager, it seemed easier to provide global metrics of cache utilization across FileIO instances.